specs/eventing/data-plane.md

```diff
@@ -1,125 +1,179 @@
-# Knative Eventing Data Plane Contracts
+# Knative Eventing Data Plane Contract
 
 ## Introduction
 
-Developers using Knative Eventing need to know what is supported for delivery to
-user provided components that receive events. Knative Eventing defines contract
-for data plane components and we have listed them here.
+Late-binding event senders and receivers (composing applications using
+configuration) only works when all event senders and recipients speak a common
+protocol. In order to enable wide support for senders and receivers, Knative
+Eventing extends the
+[CloudEvents HTTP bindings](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md)
+with additional semantics for the following reasons:
 
-## Conformance
+- Knative Eventing aims to enable at least once event processing; hence it
+  prefers duplicate delivery to discarded events. The CloudEvents spec does not
+  take a stance here.
 
-The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
-"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
-interpreted as described in RFC2119.
-
-## Data plane contract for Sinks
+- The CloudEvents HTTP bindings provide a relatively simple and efficient
+  network protocol which can easily be supported in a wide variety of
+  programming languages leveraging existing library investments in HTTP. The
+  CloudEvents project has already written these libraries for many popular
+  languages.
 
-A **Sink** MUST be able to handle duplicate events.
+- Knative Eventing assumes a sender-driven (push) event delivery system. That
+  is, each event processor is actively responsible for an event until it is
+  handled (or affirmatively delivered to all following recipients).
 
-A **Sink** is an [_addressable_](./interfaces.md#addressable) resource that
-takes responsibility for the event. A Sink could be a consumer of events, or
-middleware. A Sink MUST be able to receive CloudEvents over HTTP and HTTPS.
-
-A **Sink** MAY be [_callable_](./interfaces.md#callable) resource that
-represents an Addressable endpoint which receives an event as input and
-optionally returns an event to forward downstream.
-
-Almost every component in Knative Eventing may be a Sink providing
-composability.
-
-Every Sink MUST support HTTP Protocol Binding for CloudEvents
-[version 1.0](https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md)
-and
-[version 0.3](https://github.com/cloudevents/spec/blob/v0.3/http-transport-binding.md)
-with restrictions and extensions specified below.
+- Knative Eventing aims to make writing
+  [event sources](./overview.md#event-source) and event-processing software
```

Comment:
let's use one term for sender and one or receiver.  Or use "source" and "event processor". But we shouldn't mix them,it'll just confuse things